### PR TITLE
Associations

### PR DESCRIPTION
**Pull request by rpire**
## Changes made:
- Added the files , `classroom.rb`, `book.rb`, and `rental.rb`. Introducing the classes, `Classroom`, `Book`, and `Rental` respectively.
- Included the `classroom` method in the `Student` class.
- Included the `rent` method to the `Person` and `Book` classes.

### CODE REVIEW

File: book.rb [hunk 1]
```diff
@@ -0,0 +1,17 @@
...
+class Book
+  def initialize(title, author)
+    @title = title
+    @author = author
+    @rentals = []
+  end
+
+  attr_accessor :title, :book
+
+  attr_reader :rentals
```
Comment (mikethreels) [inline book.rb]: -  [x] great job on setting up your associations 👏 in this line you're setting up the `attr_accessor` for `@title` and `@book` I think you might have made a typo here as the `attr_accessor` should be set for `@author` and `@title`, for `@book` it is not necessary.
- [x] for `@rentals` you might want to setup both the setter and the getter as well.

File: rental.rb [hunk 2]
```diff
@@ -0,0 +1,15 @@
...
+    @book = book
+    book.rentals.push(self)
+
+    @person = person
+    person.rentals.push(self)
+  end
+
+  attr_accessor :date
+
+  attr_reader :book, :person
```
Comment (mikethreels) [inline rental.rb:14]: - [x] great job on the associations 👍  for `@book` and `@person` you might want to set up both the setter and the getter as well. you can refer to [this article](https://github.com/microverseinc/curriculum-ruby/blob/main/oop/articles/oop_principles_examples.md#useful-shortcuts-attr_reader-attr_writer-and-attr_accessor) for more details.

File: person.rb [hunk 3]
```diff
@@ -8,13 +9,14 @@ def initialize(age, name = 'Unknown', parent_permission = DEFAULT_BOOL)
...
     @parent_permission = parent_permission
+    @rentals = []
 
     @corrector = Corrector.new
   end
 
   attr_accessor :name, :age
 
-  attr_reader :id
+  attr_reader :id, :rentals
```
Comment (mikethreels) [inline person.rb:19]: - [x] for `@rentals` you might want to set up both the setter and the getter as well.

**Review by mikethreels — CHANGES_REQUESTED** [review]
Hi Rubén, 

Good job so far!
There are some issues that you still need to work on to go to the next project but you are almost there!

## Required Changes ♻️
_Check the comments under the review._

Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561) you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

**Comment by rpire** [comment]
Hello, @mikethreels 👋

Thank you for reviewing my work. I agree with most of the required changes, but there is one that has kept me thinking.

You suggest setting up a setter and a getter for the `rentals`  attribute in both `Person` and `Book` classes. But that will make the value of `rentals` mutable by simply doing something like: `person_a.rentals = 'Happy Coding!'`.

And also, a change made this way wouldn't be reflected in the books which that person rented. Thus my decision of giving only a reader to the `rentals` attribute in both classes. This way I would avoid altering that attribute in a way that might break the program and only change the value of `rentals` through the creation of a new instance of `Rental` or the `rent` method.

On top of all this, giving a setter to the `rentals` attribute would allow the value to be set to something different than an `array`. That would cause problems to create more instances of the `Rental` class since it depends on pushing values into the `rentals` `array`, which would now be a `string`.

Being all said. Will it still be a good idea to give the `rentals` attribute a setter? I sincerely believe it's better to give only a reader to that attribute in particular.

Thank you in advance for your help. I will start working on the rest of the required changes. ⚒️

 Cheers and Happy coding to you too. 🤓

> EDIT: I noticed after uploading this comment but, I would have the same problem if I give a setter to `book` and `person` attributes in the `Rental` class. Plus, requirements only mention giving setter and getter to the `date` attribute. I believe only readers for `book` and `person` would work just fine. What do you think?

**Review by rloterh — APPROVED** [review]
## Status: Approved :heavy_check_mark:

Hi Rubén, 

Your project is complete! Great job for making the necessary changes recommended in the previous review. There is nothing else to say other than... it's time to merge it 🍾 :shipit:.
Congratulations! 🎉🥂

<br>
Cheers and Happy coding!👏 💻 🍷 

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561) you have 1 more limited review per this project. If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

**Comment by rpire** [comment]
Hello, @rloterh 👋

Thank you for reviewing my work. I will proceed to merge these changes into the `dev` branch. 🚀

Cheers and Happy coding to you too! 🤓

**Pull request merged**